From f3a6681d075ef7fdb4f1cf75c1cf406e9929740f Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Mon, 9 Apr 2018 10:07:54 +0100
Subject: [PATCH] uscan: Handle absolute USCAN_DESTDIR in git mode

An absolute path for USCAN_DESTDIR was handled everywhere else, but in
git mode uscan blindly appended it to $curdir, resulting in paths
like "/home/smcv/src/debian/ioquake3//home/smcv/tmp/build-area..."
when used.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 scripts/uscan.pl | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/scripts/uscan.pl b/scripts/uscan.pl
index 06428164..bd7d693b 100755
--- a/scripts/uscan.pl
+++ b/scripts/uscan.pl
@@ -4435,6 +4435,7 @@ sub downloader ($$$$$)
 	my $curdir = cwd();
 	$fname =~ m%(.*)/([^/]*)-([^_/-]*)\.tar\.(gz|xz|bz2|lzma)%;
 	my $dst = $1;
+	my $abs_dst = abs_path($dst);
 	my $pkg = $2;
 	my $ver = $3;
 	my $suffix = $4;
@@ -4451,14 +4452,14 @@ sub downloader ($$$$$)
 		$gitrepo_state=2;
 	    }
 	}
-	uscan_verbose "Execute: git --git-dir=$destdir/$gitrepo_dir archive --format=tar --prefix=$pkg-$ver/ --output=$curdir/$dst/$pkg-$ver.tar $gitref\n";
-	system('git', "--git-dir=$destdir/$gitrepo_dir", 'archive', '--format=tar', "--prefix=$pkg-$ver/", "--output=$curdir/$dst/$pkg-$ver.tar", $gitref) == 0 or uscan_die("git archive failed\n");
+	uscan_verbose "Execute: git --git-dir=$destdir/$gitrepo_dir archive --format=tar --prefix=$pkg-$ver/ --output=$abs_dst/$pkg-$ver.tar $gitref\n";
+	system('git', "--git-dir=$destdir/$gitrepo_dir", 'archive', '--format=tar', "--prefix=$pkg-$ver/", "--output=$abs_dst/$pkg-$ver.tar", $gitref) == 0 or uscan_die("git archive failed\n");
 	# If git cloned repo exists and not --debug ($verbose=1) -> remove it
 	if ($gitrepo_state > 0 and $verbose < 1) {
-	    system('rm', '-rf', "$curdir/$dst/$gitrepo_dir");
+	    system('rm', '-rf', "$abs_dst/$gitrepo_dir");
 	    $gitrepo_state=0;
 	}
-	chdir "$curdir/$dst" or uscan_die("Unable to chdir($curdir/$dst): $!\n");
+	chdir "$abs_dst" or uscan_die("Unable to chdir($abs_dst): $!\n");
 	if ($suffix eq 'gz') {
 	    uscan_verbose "Execute: gzip -n -9 $pkg-$ver.tar\n";
 	    system("gzip", "-n", "-9", "$pkg-$ver.tar") == 0 or uscan_die("gzip failed\n");
-- 
2.17.0

